src/JSHandle.ts

```diff
@@ -467,10 +497,18 @@ export class ElementHandle extends JSHandle {
   }
 
   async $$eval<ReturnType extends any>(selector: string, pageFunction: Function | string, ...args: unknown[]): Promise<ReturnType> {
-    const arrayHandle = await this.evaluateHandle(
-        (element, selector) => Array.from(element.querySelectorAll(selector)),
-        selector
-    );
+    let handler = (element: Element, selector: string): JSHandle => Array.from(element.querySelectorAll(selector)) as unknown as JSHandle;
+    if (selector.indexOf(SEARCH_SEPARATOR) !== -1) {
+      const {name, updatedSelector} = this._getQueryFunctionNameAndSelector(selector);
+      const queryFunction = customQueryHandlers().get(name);
+      if (!queryFunction)
+        throw new Error(`$$ query set to use "${name}", but no query function of that name was found`);
```

Comment:
`$$` -> `$$eval`, but we don't need to be this specific here. If we abstract this into a shared helper between `$$eval` and `$$` as suggested above this will automatically be resolved